src/collective/lazysizes/tests/test_transformer.py

```diff
@@ -88,3 +95,91 @@ def test_lazyload_tweet_no_script(self):
         element = lxml.html.fromstring(TWEET_NO_SCRIPT)
         # the transformer returns None (skip element)
         self.assertIsNone(self.transformer._lazyload_tweet(element))
+
+    def test_lazyload_availability(self):
+        """Test, if transform is available.
+        """
+        tpl = u'<html><body><div id="content">'\
+            + u'<img src="{0}"/>'\
+            + u'</div></body></html>'
+        url = u'http://example.com/foo.png'
+        html = tpl.format(url)
+
+        # Case 1: No transform for logged-in
+        result = self.transformer.transformIterable(html, 'utf-8')
+        self.assertEqual(result, None)
+
+        # LOGOUT for following tests
+        logout()
+
+        # Case 2: Successful transform for anonymous
+        result = self.transformer.transformIterable(html, 'utf-8')
+        img = result.tree.xpath('//img')[0]
+        self.assertEqual(img.attrib['data-src'], url)
+        self.assertEqual(img.attrib['src'], PLACEHOLDER)
+
+    def test_lazyload_custom_availability(self):
+        """Test custom availability.
+        """
+        tpl = u'<html><body><div id="content">'\
+            + u'<img src="{0}"/>'\
+            + u'</div></body></html>'
+        url = u'http://example.com/foo.png'
+        html = tpl.format(url)
+
+        # Instantiate transformer - we're going to modify the instance.
+        transformer = LazySizesTransform(None, self.layer['request'])
+
+        # We're logged in, right?
+        self.assertTrue(plone.api.user.is_anonymous() is False)
+
+        # By default, the transform isn't available for logged in users.
+        result = transformer.transformIterable(html, 'utf-8')
+        self.assertEqual(result, None)
+
+        # After providing a custom ``available`` method it is.
+        # NOTE: You should subclass and register the transform, not monkey
+        # patching like here.
+        transformer.available = lambda: True
+        result = transformer.transformIterable(html, 'utf-8')
+        img = result.tree.xpath('//img')[0]
+        self.assertEqual(img.attrib['data-src'], url)
+        self.assertEqual(img.attrib['src'], PLACEHOLDER)
+
+    def test_lazyload_blacklist(self):
+        """Test, if blacklisted elements are transformed.
+        """
+        registry = getUtility(IRegistry)
+        settings = registry.forInterface(ILazySizesSettings)
+        settings.css_class_blacklist = set(['nolazyload'])
+
+        tpl = u'<html><body><div id="content">'\
+            + u'<img src="{0}" class="{1}"/>'\
+            + u'</div></body></html>'
+        url = u'http://example.com/foo.png'
+
+        logout()  # Make available
+
+        # Case 1: Do not transform Blacklisted - single class
+        html = tpl.format(url, 'nolazyload')
+        result = self.transformer.transformIterable(html, 'utf-8')
+        img = result.tree.xpath('//img')[0]
+        self.assertTrue('data-src' not in img.attrib.keys())
```

Comment:
use `self.assertNotIn()` instead.

Reply:
I actually prefer assertTrue because the evaluation expression is easier to read.
But I understand, the test failure output is less informative.
Will change with next iteration.

Reply:
in fact is the opposite: using `assertTrue` forces you to first evaluate the assertion and then evaluate the expression itself; that's a two-step mental process; see:

```python
self.assertTrue('data-src' not in img.attrib.keys())
self.assertNotIn('data-src', img.attrib)  # I think there's no need for .keys() neither
```

same happens with this, for instance:

```python
self.assertTrue(foo is None)
self.assertIsNone(foo)
```